### ISSUE TITLE
>=0.7.0, some connection keys in objects returned from Relay "malformed" 

### ISSUE THREAD

**Issue by zth** [facebook/relay#883]
Hi!

I noticed an issue with the keys of some of my connections I get back from Relay when going from 0.6.1 to 0.7.0 (also appears in 0.7.1). Basically, what happens is the following:

In my container, I query for data with a few levels of nesting and various connections. Example (with a lot of variables omitted):
```
...
teams(first: 2000) {
  count
  edges {
    node {
      surveys(first: 2000) {
        count
        edges {
          node {
            id
          }
        }
      }
    }
  }
}
```
This used to give me a nice object with edges and nodes as expected that I could loop through and extract what data I needed. However, in Relay 0.7.0 and greater the key of _surveys_ comes back "malformed" (probably not the right word...), for instance like **_surveys1UzlHL**.  I did a little digging with the queries etc and this looks like some form of internal key for separating queries/fragments that query the same types but in different components...? Anyway, I guess my question is: Am I doing something wrong here, or am I missing something? Or is this perhaps a bug?

Please let me know if I can provide any more information to help. Like I said, it doesn't happen in 0.6.1 so it was introduced (for me at least) in 0.7.0, and 0.7.1 gives the same result.

Thanks for your time.

**Comment by josephsavona** [comment]
@zth Can you clarify exactly where you are accessing the data such that they keys are different than you would expect? I.e. are you seeing the `_surveys1UzlHL`-style  key in the network response or in the props passed to your component? 

Note that Relay auto-aliases fields when it requests data from the server, but always uses the original name when vending data to components (i.e. we would expect for this type of key to appear in the network response, we would not expect to see it in your props).

**Comment by zth** [comment]
Thanks for your quick reply! I get it in the props passed to my component. I get it in the network request too of course, but that I understand and I figured it was something like aliasing. But yeah, it's in my props instead of "surveys" which is what I expected and get pre-0.7.0. 

All the data I queried for and expected to get is still there too in the props (including the data nested under "surveys" but with the aliased key), it's just a few keys that are weird and by that making the data inaccessible in a reasonable way for me.

**Comment by josephsavona** [comment]
@zth Can you provide the full container fragment definition and the props that it is receiving?

**Comment by zth** [comment]
Sure! Here's the full fragment:

```
user: () => Relay.QL`
      fragment on User {
        _id
        id
        teams(first: 2000) {
          count
          edges {
            node {
              surveys(first: 20000) {
                count
                edges {
                  node {
                    id
                    _id
                    title
                    slug
                    description
                    surveyType {
                      id
                      title
                    }
                  }
                }
              }
            }
          }
        }
      }
    `
```

And here're the props I'm getting:
```
{
  "__dataID__": "VXNlcjo1NmNkZjdjYmQ0MzY5NWJjNTdlZDFmY2I=",
  "_id": "56cdf7cbd43695bc57ed1fcb",
  "id": "VXNlcjo1NmNkZjdjYmQ0MzY5NWJjNTdlZDFmY2I=",
  "teams": {
    "count": 1,
    "edges": [
      {
        "node": {
          "_surveys1UzlHL": {
            "count": 2,
            "edges": [
              {
                "node": {
                  "id": "U3VydmV5OjU2Y2RmN2NiZDQzNjk1YmM1N2VkMWZiNA==",
                  "_id": "56cdf7cbd43695bc57ed1fb4",
                  "title": "Example survey 1",
                  "slug": "example-survey-1",
                  "description": "Sample survey.",
                  "surveyType": {
                    "id": "U3VydmV5VHlwZTo1NmNkZjdjYmQ0MzY5NWJjNTdlZDFmYmM=",
                    "title": "examplesurvey"
                  }
                },
                "cursor": "Y29ubmVjdGlvbi41NmNkZjdjYmQ0MzY5NWJjNTdlZDFmYjQ="
              },
              {
                "node": {
                  "id": "U3VydmV5OjU2Y2RmN2NiZDQzNjk1YmM1N2VkMWZiNQ==",
                  "_id": "56cdf7cbd43695bc57ed1fb5",
                  "title": "Example survey 2",
                  "slug": "example-survey-2",
                  "description": "Example survey.",
                  "surveyType": {
                    "id": "U3VydmV5VHlwZTo1NmNkZjdjYmQ0MzY5NWJjNTdlZDFmYmM=",
                    "title": "examplesurvey"
                  }
                },
                "cursor": "Y29ubmVjdGlvbi41NmNkZjdjYmQ0MzY5NWJjNTdlZDFmYjU="
              }
            ],
            "pageInfo": {
              "hasNextPage": false,
              "hasPreviousPage": false
            }
          },
          "id": "VGVhbTo1NmNkZjdjYmQ0MzY5NWJjNTdlZDFmYzQ="
        },
        "cursor": "Y29ubmVjdGlvbi41NmNkZjdjYmQ0MzY5NWJjNTdlZDFmYzQ="
      }
    ],
    "pageInfo": {
      "hasNextPage": false,
      "hasPreviousPage": false
    }
  }
}
```

**Comment by josephsavona** [comment]
Hmm - this doesn't look like data that was directly provided by Relay. RelayContainer provides props that have certain opaque metadata keys - such as `__dataID__` - on all records. The data you provided only has that key at the very top, not any of the other records. Are you accessing this data directly from the network or using another API to read this data from Relay?

**Comment by zth** [comment]
The only thing I can think of that'd be involved would be react-router-relay. Could that perhaps be a factor? Nothing else should be interacting with the data before the component get it, it's a normal React component fed into Relay.createContainer, nothing fancy going on except the react-router-relay with react-router. 

This is the exact same props/object, just downgrading to Relay 0.6.1:

```
{
  "__dataID__": "VXNlcjo1NmNkZjdjYmQ0MzY5NWJjNTdlZDFmY2I=",
  "_id": "56cdf7cbd43695bc57ed1fcb",
  "id": "VXNlcjo1NmNkZjdjYmQ0MzY5NWJjNTdlZDFmY2I=",
  "teams": {
    "__dataID__": "client:17549826716_first(2000)",
    "count": 1,
    "edges": [
      {
        "__dataID__": "client:client:17549826716:VGVhbTo1NmNkZjdjYmQ0MzY5NWJjNTdlZDFmYzQ=",
        "node": {
          "__dataID__": "VGVhbTo1NmNkZjdjYmQ0MzY5NWJjNTdlZDFmYzQ=",
          "surveys": {
            "__dataID__": "client:17549826717_first(20000)",
            "count": 2,
            "edges": [
              {
                "__dataID__": "client:client:17549826717:U3VydmV5OjU2Y2RmN2NiZDQzNjk1YmM1N2VkMWZiNA==",
                "node": {
                  "__dataID__": "U3VydmV5OjU2Y2RmN2NiZDQzNjk1YmM1N2VkMWZiNA==",
                  "id": "U3VydmV5OjU2Y2RmN2NiZDQzNjk1YmM1N2VkMWZiNA==",
                  "_id": "56cdf7cbd43695bc57ed1fb4",
                  "title": "Example survey 1",
                  "slug": "example-survey-1",
                  "description": "Sample survey.",
                  "surveyType": {
                    "__dataID__": "U3VydmV5VHlwZTo1NmNkZjdjYmQ0MzY5NWJjNTdlZDFmYmM=",
                    "id": "U3VydmV5VHlwZTo1NmNkZjdjYmQ0MzY5NWJjNTdlZDFmYmM=",
                    "title": "examplesurvey"
                  }
                }
              },
              {
                "__dataID__": "client:client:17549826717:U3VydmV5OjU2Y2RmN2NiZDQzNjk1YmM1N2VkMWZiNQ==",
                "node": {
                  "__dataID__": "U3VydmV5OjU2Y2RmN2NiZDQzNjk1YmM1N2VkMWZiNQ==",
                  "id": "U3VydmV5OjU2Y2RmN2NiZDQzNjk1YmM1N2VkMWZiNQ==",
                  "_id": "56cdf7cbd43695bc57ed1fb5",
                  "title": "Example survey 2",
                  "slug": "example-survey-2",
                  "description": "Example survey.",
                  "surveyType": {
                    "__dataID__": "U3VydmV5VHlwZTo1NmNkZjdjYmQ0MzY5NWJjNTdlZDFmYmM=",
                    "id": "U3VydmV5VHlwZTo1NmNkZjdjYmQ0MzY5NWJjNTdlZDFmYmM=",
                    "title": "examplesurvey"
                  }
                }
              }
            ]
          }
        }
      }
    ]
  }
}
```

Quite different, so I guess something's going on.

**Comment by josephsavona** [comment]
Hmm, it definitely could be due to the interaction between Relay and react-router-relay. If you render the same view directly (e.g. via a plain RelayRootContainer) does it get the expected data?

**Comment by zth** [comment]
Super weird - I threw out my node_modules, did npm install again and now it's working! It was also mentioned in the Discord chat that someone had had a similar issue a few weeks back, which was solved by making sure babel-relay-plugin was at the same version as Relay itself. I tried that with no success before posting here, but maybe that was the issue anyway and a fresh npm install fixed it?

Anyway, I'll close this issue now as it's working but I'll keep this fresh in mind if I see anyone else with similar issues. @josephsavona thank you very much for your extremely quick replies and your help!
